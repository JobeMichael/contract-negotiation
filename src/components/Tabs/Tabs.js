import React, { useContext } from "react";
import { appContext } from "../../store/appContext";
import * as Styled from "./Tabs.styles";

const tabsData = [
  { type: "employer", text: "Employer" },
  { type: "employee", text: "Employee" },
];

const Tabs = ({ children }) => {
  const {
    setAppState,
    appState: { currentUser },
  } = useContext(appContext);

  const tabClickHandler = (type) => {
    setAppState((prevState) => ({
      ...prevState,
      currentUser: type,
    }));
  };

  return (
    <>
      <Styled.ButtonGroup>
        {tabsData.map(({ type, text }) => (
          <Styled.Tab
            key={type}
            active={currentUser === type}
            onClick={() => tabClickHandler(type)}
            data-active={currentUser === type}
          >
            {text}
          </Styled.Tab>
        ))}
      </Styled.ButtonGroup>
      <p />
      <div>{children}</div>
    </>
  );
};

export default Tabs;
