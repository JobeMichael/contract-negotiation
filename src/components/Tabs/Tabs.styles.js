import styled from "styled-components";

export const Tab = styled.button`
  font-size: 20px;
  padding: 10px 60px;
  cursor: pointer;
  opacity: 0.6;
  background: white;
  border: 0;
  outline: 0;
  width: 50%;
  ${({ active }) =>
    active &&
    `
    border-bottom: 2px solid #29bdba;
    opacity: 1;
  `}
`;

export const ButtonGroup = styled.div`
  display: flex;
`;
