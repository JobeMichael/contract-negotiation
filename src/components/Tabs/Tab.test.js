import { act, cleanup, fireEvent, render } from "@testing-library/react";
import React from "react";
import { appContext } from "../../store/appContext";
import Tabs from "./Tabs";

describe("<NegotiationForm/>", () => {
  const setAppState = jest.fn();

  afterEach(() => {
    setAppState.mockRestore();
    cleanup();
  });

  let renderResult;

  const mockContext = {
    appState: {
      currentUser: "employer",
      employerInput: 0,
      employeeInput: 0,
    },
    setAppState: setAppState,
  };

  const setup = () => (
    <appContext.Provider value={mockContext}>
      <Tabs>
        <h1>Children</h1>>
      </Tabs>
    </appContext.Provider>
  );

  it("should renders NegotiationForm properly", () => {
    act(() => {
      renderResult = render(setup());
    });

    expect(renderResult.baseElement).toMatchSnapshot();
  });

  it("should work all tabs properly", () => {
    const { getByText } = render(setup());

    const tabEmployee = getByText(/Employee/i);
    const tabEmployer = getByText(/Employer/i);

    expect(tabEmployee).toBeInTheDocument();
    expect(tabEmployer).toBeInTheDocument();

    fireEvent.click(tabEmployee);

    expect(tabEmployer.getAttribute("data-active")).toBe("true");
    expect(tabEmployee.getAttribute("data-active")).toBe("false");
  });
});
