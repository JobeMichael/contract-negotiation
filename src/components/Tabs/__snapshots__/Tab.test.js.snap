// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NegotiationForm/> should renders NegotiationForm properly 1`] = `
<body>
  <div>
    <div
      class="sc-AxirZ hlnYRa"
    >
      <button
        class="sc-AxjAm fAvqRA"
        data-active="true"
      >
        Employer
      </button>
      <button
        class="sc-AxjAm jEALMi"
        data-active="false"
      >
        Employee
      </button>
    </div>
    <p />
    <div>
      <h1>
        Children
      </h1>
      &gt;
    </div>
  </div>
</body>
`;
